### DISCUSSION TITLE
Recommendations for addPeak2GeneLinks function parameters

### DISCUSSION THREAD

**Discussion by cschmidl** [GreenleafLab/ArchR#782] [Questions / Documentation]
Dear Ryan & Jeffrey

could you give some rough recommendations for the parameters used in addPeak2GeneLinks parameters, especially on the k, knnIteration, predictionCutoff and overlapCutoff? Specifically, in what way/direction it makes sense to change the standard settings if you have e.g. very small or very large datasets, relatively similar cells (eg. from one lineage) or highly diverse cells (such as PBMCs with many very different cell types)?

Thank you very much for this wonderful tool once again!

**Comment by rcorces** [comment]
Please note that we no longer use the Issues section for questions on usage or feature requests. These belong in Discussions. Issues should only be used for Bug Reports. I have migrated your post to discussions.

It is difficult to provide precise recommendations but here is some general advice:
`k` is used for creating cell groups for other analyses. If your dataset is small, you should reduce this number to avoid grouping cells that are not highly similar
`knnIteration` - I dont see much reason to change this from the default based on dataset size
`overlapCutoff` - I think I would avoid changing this and modulate `k` instead. Though if your dataset is small (<10k cells) and the number of distinct cell types is high, peak to gene analysis might be less robust. I would try to sanity check the predictions by looking at tracks if your dataset isnt ideally suited. The bigger the dataset, the lower this value could go.
`predictionCutoff` - I think I would also avoid changing this, certainly dont think I would lower it. This doesnt really change with dataset size and is more about how well matched your RNA data is to your ATAC data.
